### PR TITLE
Retry http fetching without gzip if failed

### PR DESCRIPTION
**Pull request by martinraison**
@raymondng I had to factor the actual http fetching in a "fetchHandler" function (using a convenience case class for returning all the information to the main fetch() function)

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala [hunk 1]
```diff
@@ -249,14 +254,16 @@ class HttpFetcherImpl(val airbrake:AirbrakeNotifier, userAgent: String, connecti
     httpContext.setAttribute("scraper_destination_url", url)
     httpContext.setAttribute("redirects", Seq.empty[HttpRedirect])
 
-    val fetchInfo = FetchInfo(url, System.currentTimeMillis, httpGet, Thread.currentThread()) // pass this up
-    val responseOpt = try {
+    if (disableGzip) httpGet.addHeader(ACCEPT_ENCODING, "")
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala:257]: Cool. Does this do the trick? Or do we still fail?

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala:257]: So I just manually tested with gzip enabled and disabled, but I didn't test the scenario where it first fails and then succeeds - do we have a specific example url for this?

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala:257]: (seems like http://www.sratim.co.il/view.php?id=33893&q=the+office could do the trick - testing it)

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala:257]: You can pick from any recent failures :)

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala:257]: I'm not observing the ZipException when I try to keep this page. I also tried with the other urls causing this error ...maybe the setting is not exactly the same in dev and in prod and/or the gzipping problem has somehow been fixed in the meantime.

(btw I only saw about 3-4 failures on Airbrake - were there more of them somewhere else?)

File: kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala [hunk 2]
```diff
@@ -249,14 +254,16 @@ class HttpFetcherImpl(val airbrake:AirbrakeNotifier, userAgent: String, connecti
...
+    implicit val fetchInfo = FetchInfo(url, System.currentTimeMillis, httpGet, Thread.currentThread()) // pass this up
+    try {
       q.offer(WeakReference(fetchInfo))
       val response = timingWithResult[CloseableHttpResponse](s"fetch($url).execute", { r:CloseableHttpResponse => r.getStatusLine.toString }) { httpClient.execute(httpGet, httpContext) }
       fetchInfo.respStatusRef.set(response.getStatusLine)
       Some(response)
     } catch {
-        case e:ZipException => logAndSet(fetchInfo, None)(e, "fetch", url) // todo: retry without gzip
+        case e:ZipException => if (disableGzip) logAndSet(fetchInfo, None)(e, "fetch", url)
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala]: We should be notified whether it's failing after disabling gzip. We can either send out an email or airbrake (add true to the end).

File: kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala [hunk 3]
```diff
@@ -224,10 +224,15 @@ class HttpFetcherImpl(val airbrake:AirbrakeNotifier, userAgent: String, connecti
   })
   scheduler.scheduleWithFixedDelay(enforcer, 30, 10, TimeUnit.SECONDS)
 
+  private case class HttpTransaction(responseOpt: Option[CloseableHttpResponse], fetchInfo: FetchInfo, httpGet: HttpGet, httpContext: HttpContext)
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/scraper/HttpFetcher.scala]: Transaction is a bit of a loaded term -- can you think of something else?

**Pull request merged**